paramiko/sftp_client.py

```diff
@@ -740,12 +740,10 @@ def _async_request(self, fileobj, t, *arg):
                     msg.add_int64(item)
                 elif isinstance(item, int):
                     msg.add_int(item)
-                elif isinstance(item, (string_types, bytes_types)):
-                    msg.add_string(item)
                 elif isinstance(item, SFTPAttributes):
                     item._pack(msg)
                 else:
-                    raise Exception('unknown type for %r type %r' % (item, type(item)))
```

Comment:
I'd love a comment here noting explicitly how "it was not any of the other expected types, so we assume it's something that can be `asbytes()`'d into a packet-level string type, and hand it to `add_string()` which does so" (assuming I'm interpreting that intent correctly!)